core/capabilities/ccip/configs/solana/chain_writer.go

```diff
@@ -0,0 +1,330 @@
+				},
+			},
+			chainwriter.PDALookups{
+				Name: "PerChainTokenConfig",
+				PublicKey: chainwriter.AccountConstant{
+					Address: routerProgramAddress,
+				},
+				Seeds: []chainwriter.Seed{
+					{Dynamic: chainwriter.AccountLookup{Location: destTokenAddress}},
+					{Dynamic: chainwriter.AccountLookup{Location: destChainSelectorPath}},
+				},
+				IsSigner:   false,
+				IsWritable: false,
+			},
+			chainwriter.AccountsFromLookupTable{
+				LookupTableName: "RegistryTokenState",
+				IncludeIndexes:  []int{},
+			},
+			chainwriter.PDALookups{
+				Name: "RegistryTokenConfig",
+				PublicKey: chainwriter.AccountConstant{
+					Address:    routerProgramAddress,
+					IsSigner:   false,
+					IsWritable: false,
+				},
+				Seeds: []chainwriter.Seed{
+					{Dynamic: chainwriter.AccountLookup{Location: destTokenAddress}},
+				},
+				IsSigner:   false,
+				IsWritable: false,
+			},
+			chainwriter.PDALookups{
+				Name: "UserNoncePerChain",
+				PublicKey: chainwriter.AccountConstant{
+					Address:    routerProgramAddress,
+					IsSigner:   false,
+					IsWritable: false,
+				},
+				Seeds: []chainwriter.Seed{
+					{Dynamic: chainwriter.AccountLookup{Location: "Message.Receiver"}},
+					{Dynamic: chainwriter.AccountLookup{Location: destChainSelectorPath}},
+				},
+			},
+			chainwriter.PDALookups{
+				Name: "CPISigner",
+				PublicKey: chainwriter.AccountConstant{
+					Address:    routerProgramAddress,
+					IsSigner:   false,
+					IsWritable: false,
+				},
+				Seeds: []chainwriter.Seed{
+					{Static: []byte("external_token_pools_signer")},
+				},
+				IsSigner:   false,
+				IsWritable: false,
+			},
+			chainwriter.AccountConstant{
+				Name:       "ComputeBudgetProgram",
+				Address:    computeBudgetProgramAddress,
+				IsSigner:   true,
+				IsWritable: false,
+			},
+		},
+		DebugIDLocation: "Message.MessageID",
+	}
+}
+
+func GetSolanaChainWriterConfig(routerProgramAddress string, commonAddressesLookupTable solana.PublicKey, fromAddress string) (chainwriter.ChainWriterConfig, error) {
+	computeBudgetProgramAddress := solana.ComputeBudget.String()
+	sysvarInstructionsAddress := solana.SysVarInstructionsPubkey.String()
+
+	// check fromAddress
+	_, err := solana.PublicKeyFromBase58(fromAddress)
+	if err != nil {
+		return chainwriter.ChainWriterConfig{}, fmt.Errorf("invalid from address %s: %w", fromAddress, err)
+	}
+
+	// validate CCIP Router IDL, errors not expected
+	var idl codec.IDL
+	if err = json.Unmarshal([]byte(ccipRouterIDL), &idl); err != nil {
+		return chainwriter.ChainWriterConfig{}, fmt.Errorf("unexpected error: invalid CCIP Router IDL, error: %w", err)
+	}
+
+	routeAccountConfig := chainwriter.PDALookups{
+		Name: "RouterAccountConfig",
+		PublicKey: chainwriter.AccountConstant{
+			Address: routerProgramAddress,
+		},
+		Seeds: []chainwriter.Seed{
+			{Static: []byte("config")},
+		},
+		IsSigner:   false,
+		IsWritable: false,
+	}
```

Comment:
Couldn't we move this into the execute and commit method configs directly? Seems like the only thing you need is the `routerProgramAddress` which we pass into each method anyways. Also noticed the commit method config uses this account config but also separately sets `RouterAccountConfig`.

Reply:
Yeah this is confusing, I don't have this until the CI complain too many duplicate code

Reply:
Ah gotcha, SonarQube not liking it. I don't think that's a mandatory check so we could still make the call on how we like this but I don't mind leave this too. The only thing then is to remove the other instance of `RouterAccountConfig` in the commit method config.